File: assets/codes/leetcode/src/main/java/leetcode/L_35.go
package sample

func searchInsert(nums []int, target int) int {
	left := 0
	right := len(nums) - 1

	if target > nums[len(nums)-1] {
		return len(nums)
	}
	var mark int

	for left <= right {
		mid := (right-left)/2 + left
		if nums[mid] >= target {
			mark = mid
			right = mid - 1
		} else if nums[mid] < target {
			left = mid + 1
		}
	}

	return mark
}

File: assets/codes/leetcode/src/main/java/leetcode/L_35.java
package leetcode;

public class L_35 {
    public int searchInsert(int[] nums, int target) {
        int start = 0;
        int end = nums.length;
        while ((end - start) > 1) {
            int mid = (start + end) / 2;
            if (nums[mid] < target) {
                start = mid;
            } else {
                end = mid;
            }
        }
        if (nums[start] >= target) {
            return start;
        }
        return end;
    }


    public static void main(String[] args) {
        L_35 l35 = new L_35();
        System.out.println(l35.searchInsert(new int[]{1, 3, 5, 6}, 0)); // 0
        System.out.println(l35.searchInsert(new int[]{1, 3, 5, 6}, 5)); // 2
//        System.out.println(l35.searchInsert(new int[]{1, 3, 5, 6}, 2));
//        System.out.println(l35.searchInsert(new int[]{1, 3, 5, 6}, 7));
//        System.out.println(l35.searchInsert(new int[]{1}, 7));
//        System.out.println(l35.searchInsert(new int[]{1, 8}, 7));
//        System.out.println(l35.searchInsert(new int[]{1, 7}, 7));
    }
}
